validator/rpc/handler_wallet.go

```diff
@@ -166,17 +204,20 @@ func (s *Server) RecoverWallet(ctx context.Context, req *pb.RecoverWalletRequest
 		"spanish":             wordlists.Spanish,
 	}
 	if _, ok := allowedLanguages[language]; !ok {
-		return nil, status.Error(codes.InvalidArgument, "input not in the list of supported languages")
+		http2.HandleError(w, "input not in the list of supported languages", http.StatusBadRequest)
+		return
 	}
 	bip39.SetWordList(allowedLanguages[language])
 	mnemonic := req.Mnemonic
 	if err := accounts.ValidateMnemonic(mnemonic); err != nil {
-		return nil, status.Error(codes.InvalidArgument, "invalid mnemonic in request")
+		http2.HandleError(w, "invalid mnemonic in request: "+err.Error(), http.StatusBadRequest)
+		return
 	}
 
 	// Check it is not null and not an empty string.
 	if req.Mnemonic25ThWord != "" && strings.TrimSpace(req.Mnemonic25ThWord) == "" {
```

Comment:
This condition is wrong because it allows empty strings. We should keep the second check only, it will cover for both empty and whitespace.

Reply:
no , it can be null ( from json) so we need to keep the check there I believe.